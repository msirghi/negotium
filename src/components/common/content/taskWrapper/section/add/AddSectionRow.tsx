import styles from './AddSectionRow.module.scss';
import { FC, useState } from 'react';
import { Button, TextField } from '@mui/material';
import { If } from '../../../../utilities/if/If';
import { Row } from '../../../../utilities/row/Row';
import { useTranslation } from 'next-i18next';

type Props = {
  onSectionSave: (title: string, orderNumber: number) => void;
};

export const AddSectionRow: FC<Props> = ({ onSectionSave }) => {
  const [editMode, setEditMode] = useState(false);
  const [fieldValue, setFieldValue] = useState('');
  const { t } = useTranslation('common');

  const onTitleClick = () => {
    setEditMode(true);
  };

  const onCancelClick = () => {
    setEditMode(false);
  };

  const onSave = () => {
    onSectionSave(fieldValue, 1);
    setEditMode(false);
    setFieldValue('');
  };

  return (
    <div>
      <If condition={!editMode}>
        <div
          className={styles.adrTitleContainer}
          onClick={onTitleClick}
          data-testid={'title'}
        >
          <span className={styles.adrTitle}>Add section</span>
        </div>
      </If>
      <If condition={editMode}>
        <TextField
          onChange={(e) => setFieldValue(e.target.value)}
          fullWidth
          size={'small'}
          placeholder={t('inputPlaceholders.nameNewSection')}
          inputProps={{ 'data-testid': 'section-name-field' }}
          sx={{ marginTop: '1rem' }}
        />
        <Row className={styles.buttonRow}>
          <Button
            onClick={onSave}
            disabled={!fieldValue}
            data-testid={'save-button'}
            color={'primary'}
            variant={'contained'}
            size={'small'}
          >
            {t('buttonLabels.addSection')}
          </Button>
          <Button
            onClick={onCancelClick}
            size={'small'}
            data-testid={'cancel-button'}
          >
            {t('buttonLabels.cancel')}
          </Button>
        </Row>
      </If>
    </div>
  );
};
