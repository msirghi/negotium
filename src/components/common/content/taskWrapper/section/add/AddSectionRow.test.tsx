import renderer from 'react-test-renderer';
import { AddSectionRow } from './AddSectionRow';
import { act, fireEvent, render } from '@testing-library/react';

describe('AddSectionRow', () => {
  const defaultProps = {
    onSectionSave: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<AddSectionRow {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render title by default', () => {
    const { getByTestId } = render(<AddSectionRow {...defaultProps} />);
    expect(getByTestId('title')).toBeInTheDocument();
  });

  it('should show input on title click', () => {
    const { getByTestId } = render(<AddSectionRow {...defaultProps} />);
    const title = getByTestId('title');

    act(() => {
      fireEvent.click(title);
    });

    expect(getByTestId('section-name-field')).toBeInTheDocument();
  });

  it('should update the title', () => {
    const { getByTestId } = render(<AddSectionRow {...defaultProps} />);
    const title = getByTestId('title');

    act(() => {
      fireEvent.click(title);
    });

    const titleField = getByTestId('section-name-field');
    act(() => {
      fireEvent.change(titleField, { target: { value: 'new value' } });
    });

    const saveButton = getByTestId('save-button');
    act(() => {
      fireEvent.click(saveButton);
    });
    expect(defaultProps.onSectionSave).toBeCalled();
  });

  it('should cancel the edit mode', () => {
    const { getByTestId, queryByTestId } = render(
      <AddSectionRow {...defaultProps} />
    );
    const title = getByTestId('title');

    act(() => {
      fireEvent.click(title);
    });

    const cancelButton = getByTestId('cancel-button');
    act(() => {
      fireEvent.click(cancelButton);
    });
    expect(queryByTestId('section-name-field')).not.toBeInTheDocument();
  });
});
