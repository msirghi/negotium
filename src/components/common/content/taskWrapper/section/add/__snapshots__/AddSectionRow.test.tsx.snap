// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSectionRow should match the snapshot 1`] = `
<div>
  <div
    className="adrTitleContainer"
    data-testid="title"
    onClick={[Function]}
  >
    <span
      className="adrTitle"
    >
      Add section
    </span>
  </div>
</div>
`;
